src/foundry/client-esm/canvas/sources/rendered-effect-source.d.mts

```diff
@@ -0,0 +1,325 @@
+    colorDim: Color,
+    colorBright: Color,
+    colorBackground?: Color,
+  ): Color;
+
+  /**
+   * @deprecated since v11
+   */
+  set preview(preview: boolean);
+
+  /**
+   * @deprecated since v11
+   */
+  get preview(): boolean;
+}
+
+declare namespace RenderedEffectSource {
+  interface RenderedEffectSourceData extends BaseEffectSource.BaseEffectSourceData {
+    /**
+     * An animation configuration for the source
+     */
+    animation: object;
+    /**
+     * A color applied to the rendered effect
+     */
+    color: number | null;
+    /**
+     * An integer seed to synchronize (or de-synchronize) animations
+     */
+    seed: number | null;
+    /**
+     * Is this source a temporary preview?
+     */
+    preview: boolean;
+  }
+
+  interface AnimationOptions {
+    /**
+     * The animation speed, from 0 to 10
+     * @defaultValue `5`
+     */
+    speed?: number;
+
+    /**
+     * The animation intensity, from 1 to 10
+     * @defaultValue `5`
+     */
+    intensity?: number;
+
+    /**
+     * Reverse the animation direction
+     * @defaultValue `false`
+     */
+    reverse?: boolean;
+  }
+
+  interface RenderedEffectSourceAnimationConfig {
+    /**
+     * The human-readable (localized) label for the animation
+     */
+    label?: string;
+    /**
+     * The animation function that runs every frame
+     */
+    animation?: Function;
+    /**
+     * A custom illumination shader used by this animation
+     */
+    illuminationShader?: AdaptiveIlluminationShader;
+    /**
+     * A custom coloration shader used by this animation
+     */
+    colorationShader?: AdaptiveColorationShader;
+    /**
+     * A custom background shader used by this animation
+     */
+    backgroundShader?: AdaptiveBackgroundShader;
+    /**
+     * A custom darkness shader used by this animation
+     */
+    darknessShader?: AdaptiveDarknessShader;
+    /**
+     * The animation seed
+     */
+    seed?: number;
+    /**
+     * The animation time
+     */
+    time?: number;
+  }
+
+  interface RenderedEffectSourceLayer {
+    /**
+     * Is this layer actively rendered?
+     */
+    active: boolean;
+    /**
+     * Do uniforms need to be reset?
+     */
+    reset: boolean;
+    /**
+     * Is this layer temporarily suppressed?
+     */
+    suppressed: boolean;
+    /**
+     * The rendered mesh for this layer
+     */
+    mesh: PointSourceMesh;
+    /**
+     * The shader instance used for the layer
+     */
+    shader: AdaptiveLightingShader;
+  }
+
+  interface RenderedEffectLayerConfig {
+    /**
+     * The default shader used by this layer
+     */
+    defaultShader: AdaptiveLightingShader;
+    /**
+     * The blend mode used by this layer
+     */
+    blendMode: PIXI.BLEND_MODES;
+  }
+
+  type Layers = {
+    background: RenderedEffectSource.RenderedEffectSourceLayer;
+    coloration: RenderedEffectSource.RenderedEffectSourceLayer;
+    illumination: RenderedEffectSource.RenderedEffectSourceLayer;
+  };
```

Comment:
Does this ever make sense to extend? If so consider an interface.

Reply:
No, this is just shorthand for the default RenderingLayers that's set at the top